coderd/provisionerdaemons.go

```diff
@@ -62,14 +63,17 @@ func (api *API) ListenProvisionerDaemon(ctx context.Context) (client proto.DRPCP
 		}
 	}()
 
+	// Required for randomly generated names to not conflict!
+	rand.Seed(time.Now().UnixMicro())
```

Comment:
Suggestion (non-blocking): Consider moving this to early startup/init (e.g. `func main()`) or use a new rand source. Reason being that the whole process shares the same rnd, and calling `rand.Seed` multiple times and/or in multiple places can have unexpected consequences. Can then also be utilized in tests (static seed) for repeatable output.

Since this is the only use `math/rand`, this is purely future-proofing.